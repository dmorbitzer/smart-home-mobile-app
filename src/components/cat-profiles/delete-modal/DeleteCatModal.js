import { useState } from "react";
import { Modal, StyleSheet, View } from "react-native";

import DeleteCatAsk from "./DeleteCatAsk";
import DeleteCatFeedback from "./DeleteCatFeedback";

export default function DeleteCatModal(props) {
  const onCloseModal = () => {
    props.navigation.navigate("Katzenprofil", {
      refresh: true,
    });
  };
  const chooseDeleteCat = (choose) => {
    if (choose) {
      props.delteCat(props.delteId);
      setSwitchContent(
        <DeleteCatFeedback
          onCloseModal={onCloseModal}
          centeredView={styles.centeredView}
          modalView={styles.modalView}
          modalHeader={styles.modalHeader}
        />
      );
    } else {
      props.closeModalHandler(true);
    }
  };
  const [switchContent, setSwitchContent] = useState(
    <DeleteCatAsk
      chooseDeleteCat={chooseDeleteCat}
      centeredView={styles.centeredView}
      modalView={styles.modalView}
      modalHeader={styles.modalHeader}
    />
  );
  return (
    <View style={styles.centeredView}>
      <Modal animationType="slide" transparent visible={props.showModal}>
        {switchContent}
      </Modal>
    </View>
  );
}

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    marginTop: 22,
  },
  modalView: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    width: "80%",
  },
  modalHeader: {
    fontWeight: "bold",
    fontSize: 20,
    marginBottom: 10,
  },
});
