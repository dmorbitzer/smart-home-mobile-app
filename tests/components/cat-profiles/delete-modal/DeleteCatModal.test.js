import React from "react";
import renderer from "react-test-renderer";

import DeleteCatModal from "../../../../src/components/cat-profiles/delete-modal/DeleteCatModal";
jest.useFakeTimers();
jest.mock("react-native", () => {
  const RN = jest.requireActual("react-native");
  RN.Animated.timing = () => ({ start: () => {} });
  return RN;
});
const closeModalHandler = () => {
  console.log("done");
};
const showModal = true;
const props = {
  navigation: null,
  deleteCat: null,
  data: {
    cat: {
      id: "test",
    },
  },
};
test("renders DeleteCatModal correctly", () => {
  const tree = renderer
    .create(
      <DeleteCatModal
        navigation={props.navigation}
        delteCat={props.delteCat}
        delteId={props.data.cat.id}
        showModal={showModal}
        closeModalHandler={closeModalHandler}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
