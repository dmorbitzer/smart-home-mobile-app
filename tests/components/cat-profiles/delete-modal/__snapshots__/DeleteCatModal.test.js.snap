// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DeleteCatModal correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={true}
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "height": "30%",
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "bold",
              "marginBottom": 20,
            }
          }
        >
          Sind Sie sich sicher, dass Sie das Profil unwiderruflich l√∂schen wollen?
        </Text>
        <View
          style={
            [
              {
                "backgroundColor": undefined,
                "borderBottomColor": undefined,
                "borderBottomEndRadius": undefined,
                "borderBottomLeftRadius": undefined,
                "borderBottomRightRadius": undefined,
                "borderBottomStartRadius": undefined,
                "borderBottomWidth": undefined,
                "borderColor": undefined,
                "borderEndColor": undefined,
                "borderEndWidth": undefined,
                "borderLeftColor": undefined,
                "borderLeftWidth": undefined,
                "borderRadius": undefined,
                "borderRightColor": undefined,
                "borderRightWidth": undefined,
                "borderStartColor": undefined,
                "borderStartWidth": undefined,
                "borderStyle": undefined,
                "borderTopColor": undefined,
                "borderTopEndRadius": undefined,
                "borderTopLeftRadius": undefined,
                "borderTopRightRadius": undefined,
                "borderTopStartRadius": undefined,
                "borderTopWidth": undefined,
                "borderWidth": undefined,
                "bottom": undefined,
                "end": undefined,
                "height": undefined,
                "left": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginEnd": undefined,
                "marginHorizontal": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginStart": undefined,
                "marginTop": undefined,
                "marginVertical": undefined,
                "maxHeight": undefined,
                "maxWidth": undefined,
                "minHeight": undefined,
                "minWidth": undefined,
                "opacity": undefined,
                "overflow": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingEnd": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingStart": undefined,
                "paddingTop": undefined,
                "paddingVertical": undefined,
                "position": undefined,
                "right": undefined,
                "start": undefined,
                "top": undefined,
                "width": undefined,
                "zIndex": undefined,
              },
              [
                {
                  "alignContent": undefined,
                  "alignItems": undefined,
                  "alignSelf": undefined,
                  "flex": undefined,
                  "flexBasis": undefined,
                  "flexDirection": "row",
                  "flexGrow": undefined,
                  "flexShrink": undefined,
                  "flexWrap": undefined,
                  "justifyContent": undefined,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              [
                {
                  "backgroundColor": undefined,
                  "borderBottomColor": undefined,
                  "borderBottomEndRadius": undefined,
                  "borderBottomLeftRadius": undefined,
                  "borderBottomRightRadius": undefined,
                  "borderBottomStartRadius": undefined,
                  "borderBottomWidth": undefined,
                  "borderColor": undefined,
                  "borderEndColor": undefined,
                  "borderEndWidth": undefined,
                  "borderLeftColor": undefined,
                  "borderLeftWidth": undefined,
                  "borderRadius": undefined,
                  "borderRightColor": undefined,
                  "borderRightWidth": undefined,
                  "borderStartColor": undefined,
                  "borderStartWidth": undefined,
                  "borderStyle": undefined,
                  "borderTopColor": undefined,
                  "borderTopEndRadius": undefined,
                  "borderTopLeftRadius": undefined,
                  "borderTopRightRadius": undefined,
                  "borderTopStartRadius": undefined,
                  "borderTopWidth": undefined,
                  "borderWidth": undefined,
                  "bottom": undefined,
                  "end": undefined,
                  "height": undefined,
                  "left": undefined,
                  "margin": undefined,
                  "marginBottom": 0,
                  "marginEnd": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginStart": undefined,
                  "marginTop": 30,
                  "marginVertical": undefined,
                  "maxHeight": undefined,
                  "maxWidth": undefined,
                  "minHeight": undefined,
                  "minWidth": undefined,
                  "opacity": undefined,
                  "overflow": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingEnd": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingStart": undefined,
                  "paddingTop": undefined,
                  "paddingVertical": undefined,
                  "position": undefined,
                  "right": undefined,
                  "start": undefined,
                  "top": undefined,
                  "width": undefined,
                  "zIndex": undefined,
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              onTouchStart={[Function]}
              style={
                {
                  "backgroundColor": "lightblue",
                  "borderBottomEndRadius": 4,
                  "borderBottomStartRadius": 4,
                  "borderTopEndRadius": 4,
                  "borderTopStartRadius": 4,
                  "justifyContent": "center",
                  "minHeight": 60,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 1,
                    "width": 0,
                  },
                  "shadowOpacity": 0.2,
                  "shadowRadius": 1.41,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "borderBottomEndRadius": 4,
                      "borderBottomStartRadius": 4,
                      "borderTopEndRadius": 4,
                      "borderTopStartRadius": 4,
                      "overflow": "hidden",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onLayout={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 36,
                        "justifyContent": "center",
                        "minWidth": 64,
                        "paddingEnd": 16,
                        "paddingStart": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                        },
                      ]
                    }
                  />
                  <Text
                    selectable={false}
                    style={
                      [
                        {
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 1.25,
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#000000",
                        },
                        [
                          {
                            "color": "#000000",
                          },
                          {
                            "opacity": 1,
                            "textTransform": "uppercase",
                          },
                          {
                            "fontSize": 25,
                          },
                        ],
                      ]
                    }
                  >
                    Nein
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": undefined,
                  "borderBottomColor": undefined,
                  "borderBottomEndRadius": undefined,
                  "borderBottomLeftRadius": undefined,
                  "borderBottomRightRadius": undefined,
                  "borderBottomStartRadius": undefined,
                  "borderBottomWidth": undefined,
                  "borderColor": undefined,
                  "borderEndColor": undefined,
                  "borderEndWidth": undefined,
                  "borderLeftColor": undefined,
                  "borderLeftWidth": undefined,
                  "borderRadius": undefined,
                  "borderRightColor": undefined,
                  "borderRightWidth": undefined,
                  "borderStartColor": undefined,
                  "borderStartWidth": undefined,
                  "borderStyle": undefined,
                  "borderTopColor": undefined,
                  "borderTopEndRadius": undefined,
                  "borderTopLeftRadius": undefined,
                  "borderTopRightRadius": undefined,
                  "borderTopStartRadius": undefined,
                  "borderTopWidth": undefined,
                  "borderWidth": undefined,
                  "bottom": undefined,
                  "end": undefined,
                  "height": undefined,
                  "left": undefined,
                  "margin": undefined,
                  "marginBottom": 0,
                  "marginEnd": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginStart": undefined,
                  "marginTop": 30,
                  "marginVertical": undefined,
                  "maxHeight": undefined,
                  "maxWidth": undefined,
                  "minHeight": undefined,
                  "minWidth": undefined,
                  "opacity": undefined,
                  "overflow": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingEnd": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingStart": undefined,
                  "paddingTop": undefined,
                  "paddingVertical": undefined,
                  "position": undefined,
                  "right": undefined,
                  "start": undefined,
                  "top": undefined,
                  "width": undefined,
                  "zIndex": undefined,
                },
                undefined,
              ]
            }
          >
            <View
              collapsable={false}
              onTouchStart={[Function]}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderBottomEndRadius": 4,
                  "borderBottomStartRadius": 4,
                  "borderTopEndRadius": 4,
                  "borderTopStartRadius": 4,
                  "justifyContent": "center",
                  "minHeight": 60,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "borderBottomEndRadius": 4,
                      "borderBottomStartRadius": 4,
                      "borderTopEndRadius": 4,
                      "borderTopStartRadius": 4,
                      "overflow": "hidden",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onLayout={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "height": 36,
                        "justifyContent": "center",
                        "minWidth": 64,
                        "paddingEnd": 16,
                        "paddingStart": 16,
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                        {
                          "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
                        },
                      ]
                    }
                  />
                  <Text
                    selectable={false}
                    style={
                      [
                        {
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 1.25,
                          "textTransform": "uppercase",
                        },
                        {
                          "color": "#000000",
                        },
                        [
                          {
                            "color": "darkred",
                          },
                          {
                            "opacity": 1,
                            "textTransform": "uppercase",
                          },
                          {
                            "fontSize": 25,
                          },
                        ],
                      ]
                    }
                  >
                    Ja
                  </Text>
                </View>
              </View>
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    {
                      "borderBottomEndRadius": 4,
                      "borderBottomStartRadius": 4,
                      "borderColor": "#e0e0e0",
                      "borderTopEndRadius": 4,
                      "borderTopStartRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
